src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/UvWriteReq.cs

```diff
@@ -83,12 +85,14 @@ public LibuvAwaitable<UvWriteReq> WriteAsync(UvStreamHandle handle, ArraySegment
                 var index = 0;
                 foreach (var memory in buffer)
                 {
-                    var tryGetPointerResult = memory.TryGetPointer(out var pointer);
-                    Debug.Assert(tryGetPointerResult);
+                    // REVIEW: This isn't necessary for our default pool since the memory is 
+                    // already pinned but it also makes tests pass
+                    var memoryHandle = memory.Pin();
```

Comment:
That test is wrong,  passing `ReadableBuffer.Create` result which is array based and unexpected. We should just change the test.